text/0043-block-metadata.md

```diff
@@ -0,0 +1,338 @@
+
+Add a `message BlockMetadata` to `blockchain.ArchiveBlock`, and an analogous
+`struct BlockMetadata` to `mc_blockchain_types::BlockData`.
+
+## Protobuf schema
+```protobuf
+message BlockMetadataContents {
+    /// The Block ID.
+    BlockID id = 1;
+
+    /// Quorum set configuration at the time of externalization.
+    QuorumSet quorum_set = 2;
+
+    /// IAS report for the enclave which generated the signature.
+    VerificationReport verification_report = 3;
+}
+
+message BlockMetadata {
+    /// Metadata signed by the consensus node.
+    BlockMetadataContents contents = 1;
+
+    /// Message signing key.
+    Ed25519Public node_key = 2;
+
+    /// Signature using `node_key` over the Digestible encoding of `contents`.
+    Ed25519Signature signature = 3;
+}
+
+// NB: Only addition is field #4.
+message ArchiveBlockV1 {
+    /// Block.
+    Block block = 1;
+
+    /// Contents of the block.
+    BlockContents block_contents = 2;
+
+    /// Block signature.
+    BlockSignature signature = 3;
+
+    /// Additional signed metadata about this block.
+    BlockMetadata metadata = 4;
+}
+```
+
+Since this is a protobuf-additive change, we don't need a new `ArchiveBlock`
+version/type.
+
+## Signing metadata
+The metadata signature will always differ from the block signature. The block
+signature is generated by the enclave using a private key that never leaves the
+enclave, while the metadata is signed by the "untrusted" code outside the
+enclave using the SCP message signing key, since the enclave does not have this
+metadata.
+
+The metadata signature is generated from the `Digestible` encoding of
+`BlockMetadataContents`, since the protobuf wire format is not a canonical
+encoding.
+
+## Signing key history
+MobileCoin Foundation will publish a mapping/list of node_id to a range of block
+indexes, which node operators and consumers can use to verify the metadata
+signatures.
+
+This mapping will contain multiple node_id:block_range entries, in the formats defined below.
+
+#### Key History TOML format
+```toml
+[[node]]
+# Pub key, as hex-encoded or PEM-encoded data or a PEM file path.
+message_signing_pub_key = "..."
+# Index of the first block signed with this key.
+first_block_index = X
+# Optional index of the last block signed with this key.
+last_block_index = Y
+```
+
+#### Key History JSON format
+```json
+{
+  "node": [
+    {
+      "message_signing_pub_key": "...",
+      "first_block_index": 0,
+      "last_block_index": 42
+    }
+  ]
+}
+```
+
+## Historical AVRs
+The Block Signing Key used to sign the transaction data within a block is
+included within the `isvQuoteBody` field of an Intel AVR. By having AVRs
+available on chain, anyone can verify the reported block signing key is correct
+and that the block was signed by a consensus node with a valid enclave and a
+recognized SCP message signing key.
+
+Thus With `BlockMetadata`, consumers of the blockchain may not need a Watcher DB
```

Comment:
I believe the main purpose of the watcher these days other than keeping track of history is to provide guaranteed timestamps. So consumes needing block timestamps would still need it, since timestamps are coupled to block signatures and those are only available on blocks originated from an enclave and not via catch up

Reply:
Added, and moved this note about removing watcher to Future Possibilities, since it felt increasingly out of place here.